zz/macadress.sh

```diff
@@ -0,0 +1,17 @@
+# ----------------------------------------------------------------------------
+# Mostra os MAC address disponiveis.
+# Uso: zzmacadress
+# Ex.: zzmacadress
+#
+# Autor: Adriano Laureano, @sl4ureano
+# Versão: 1
+# Licença: GPL
+# ----------------------------------------------------------------------------
+zzmacadress()
+{
+    zzzz -h macadress "$1" && return
+
+    local MAC=$(ifconfig -a | grep -Po 'HWaddr \K.*$')
```

Comment:
Algumas observações:
 - Qual o significado da opção -a no ifconfig? ( realmente não sei, e não achei referência )
 - O grep está filtrando por ``` HWaddr ```, mas há saídas dependendo da versão que exibem apenas ``` HW ``` ou se for BSD será ``` ether ```.
 - No grep está usando o recurso de retrovisor com o uso do ``` \K ```, infelizmente nem todas as versões do grep suportam esse recurso. Não é citado essa situação no documento sobre portabilidade, mas na idéia original é um recurso a ser evitado.
 - Aqui vc direciona a saída para uma variável local e depois a exibe usando echo, acho que nesse caso não haveria essa necessidade, podendo ter sua saída exibida diretamente.
 - E incentivamos sempre que possível também ter um testador, para validar o resultado e prever erros.

Reply:
Obrigado pelas observações.